### PR TITLE
Simplify select overloads for words

### PR DESCRIPTION
**Pull request by JoeLoser**
###  :notebook_with_decorative_cover: Description

Simplify `select` function templates by preferring `if-constexpr` with a predicate instead of SFINAE.

###  :memo: Checklist

- [X] All user-facing changes have changelog entries.
- [X] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [X] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

File-by-file.

### CODE REVIEW

File: libvast/vast/word.hpp [hunk 1]
```diff
@@ -422,30 +422,23 @@ static constexpr auto find_prev(T x, detail::word_size_type<T> i)
...
   // TODO: make this efficient and branch-free. There is one implementation
   // that counts from the right for 64-bit here:
   // http://graphics.stanford.edu/~seander/bithacks.html
+  const auto pred = [](const auto&... args){
+    if constexpr(Bit) {
+      return word<T>::test(args...);
+    }
+    else {
+      return !word<T>::test(args...);
+    }
```
Comment (dominiklohmann) [inline libvast/vast/word.hpp:437]: We usually omit the braces if every branch is just a single statement. Not that this is necessarily better, but let's keep the style consistent throughout the code base.

```suggestion
    if constexpr(Bit)
      return word<T>::test(args...);
    else
      return !word<T>::test(args...);
```

**Review by dominiklohmann — APPROVED** [review]
Thanks for your contribution! Changes like this are always welcome. I just have one minor request regarding the code style, but other than that this looks ready to be merged.

File: libvast/vast/word.hpp (hunk 1 above)
Comment (JoeLoser) [reply libvast/vast/word.hpp]: Fair enough. It'd be good to perhaps enforce this with a `clang-tidy` check, but I just fixed it up for the purposes of this commit to be consistent. Thanks 👍

Comment (dominiklohmann) [reply libvast/vast/word.hpp]: We do have clang-tidy somewhere on our roadmap, but so far just didn't get around to doing it.

Thanks for the fixup, and thanks again for your contribution! 🎉

**Comment by JoeLoser** [comment]
CC: @dwms-qlab

**Pull request merged**